pkg/services/ngalert/state/compat.go

```diff
@@ -139,13 +140,13 @@ func errorAlert(labels, annotations data.Labels, alertState *State, urlStr strin
 	}
 }
 
-func FromStateTransitionToPostableAlerts(firingStates []StateTransition, stateManager *Manager, appURL *url.URL) apimodels.PostableAlerts {
+func FromStateTransitionToPostableAlerts(evaluatedAt time.Time, firingStates []StateTransition, stateManager *Manager, appURL *url.URL) apimodels.PostableAlerts {
 	alerts := apimodels.PostableAlerts{PostableAlerts: make([]models.PostableAlert, 0, len(firingStates))}
 	var sentAlerts []*State
-	ts := time.Now()
+	ts := evaluatedAt
```

Comment:
NOTE: This change is not strictly necessary, though I believe it is more correct than before and is useful for creating tests for this change, as other times use a mocked clock.